csvcubed/csvcubed/readers/mapcolumntocomponent.py

```diff
@@ -0,0 +1,165 @@
+        )
+    elif isinstance(schema_mapping, schema.ExistingDimension):
+        return QbColumn(
+            column_title,
+            schema_mapping.map_to_existing_qb_dimension(),
+            csv_column_uri_template=schema_mapping.value,
+        )
+    elif isinstance(schema_mapping, schema.NewAttribute):
+        return QbColumn(
+            column_title,
+            schema_mapping.map_to_new_qb_attribute(column_title, data),
+            csv_column_uri_template=schema_mapping.value,
+        )
+    elif isinstance(schema_mapping, schema.ExistingAttribute):
+        return QbColumn(
+            column_title,
+            schema_mapping.map_to_existing_qb_attribute(data),
+            csv_column_uri_template=schema_mapping.value,
+        )
+    elif isinstance(schema_mapping, schema.NewUnits):
+        return QbColumn(column_title, schema_mapping.map_to_qb_multi_units(data))
+    elif isinstance(schema_mapping, schema.ExistingUnits):
+        return QbColumn(
+            column_title,
+            schema_mapping.map_to_qb_multi_units(data, column_title),
+            csv_column_uri_template=schema_mapping.value,
+        )
+    elif isinstance(schema_mapping, schema.NewMeasures):
+        return QbColumn(
+            column_title, schema_mapping.map_to_multi_measure_dimension(data)
+        )
+    elif isinstance(schema_mapping, schema.ExistingMeasures):
+        return QbColumn(
+            column_title,
+            schema_mapping.map_to_multi_measure_dimension(column_title, data),
+            csv_column_uri_template=schema_mapping.value,
+        )
+    elif isinstance(schema_mapping, schema.ObservationValue):
+        return QbColumn(column_title, schema_mapping.map_to_qb_observation())
+    else:
+        raise ValueError(f"Unmatched schema model type {type(schema_mapping)}")
+
+
+def _from_column_dict_to_schema_model(
+    column_title: str,
+    column: dict,
+) -> Union[
+    schema.NewDimension,
+    schema.ExistingDimension,
+    schema.NewAttribute,
+    schema.ExistingAttribute,
+    schema.NewUnits,
+    schema.ExistingUnits,
+    schema.NewMeasures,
+    schema.ExistingMeasures,
+    schema.ObservationValue,
+]:
+    """
+    N.B. when using the :method:`dict_fields_match_class` method, we need to ensure that we check for types with
+    required properties *before* types without required properties.
+    """
+    column_type = column.get("type")
+    column_without_type = copy.deepcopy(column)
+    del column_without_type["type"]
+
+    if column_type is None:
+        raise ValueError("Type of column not specified.")
+    elif column_type == "dimension":
+        if any([column_without_type.get(a) for a in ['label', 'description', 'cell_uri_template', 'code_list']]):
+            # probably a New dimension, so move all properties under a new node
+            # 'new' can be bool or NewDimensionProperty (only former works)
+            column = {'type': column_type,
```

Comment:
Instead of doing this reshaping, the `schema.ExistingDimension` and `schema.NewDimension` classes should be rewritten so that they exactly match the properties available in the JSON document. 

They're designed to be classes which exactly match the style of JSON the user is expected to write.

Reply:
Let's have a call about this.

Reply:
Updates were made following discussion so they are now separate and it uses the proposed logic to choose.